contracts/escrows/CountdownGriefingEscrow.sol

```diff
@@ -0,0 +1,421 @@
+        {
+            uint256 paymentAmount = Deposit._clearDeposit(_data.buyer);
+            uint256 stakeAmount = Deposit._clearDeposit(_data.seller);
+            totalStake = paymentAmount.add(stakeAmount);
+        }
+
+        if (totalStake > 0) {
+            require(IERC20(BurnNMR.getToken()).approve(agreement, totalStake), "token approval failed");
+            CountdownGriefing(agreement).increaseStake(totalStake);
+        }
+
+        // start agreement countdown
+
+        CountdownGriefing(agreement).startCountdown();
+
+        // transfer operator
+        address operator = Operated.getOperator();
+        if (operator != address(0)) {
+            CountdownGriefing(agreement).transferOperator(operator);
+        } else {
+            CountdownGriefing(agreement).renounceOperator();
+        }
+
+        // update status
+        _data.status = EscrowStatus.isFinalized;
+
+        // delete storage
+        delete _data.paymentAmount;
+        delete _data.stakeAmount;
+        delete _data.agreementParams;
+
+        // emit event
+        emit Finalized(agreement);
+    }
+
+    /// @notice Submit data to the buyer
+    /// @dev Access Control: seller OR operator
+    ///      State Machine: after finalize()
+    function submitData(bytes memory data) public {
+        // restrict access control
+        require(isSeller(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or active operator");
+        // restrict state machine
+        require(isFinalized(), "only after finalized");
+
+        // emit event
+        emit DataSubmitted(data);
+    }
+
+    /// @notice Cancel escrow because no interested counterparty
+    ///          - return deposit to caller
+    ///          - close escrow
+    /// @dev Access Control: seller OR buyer OR operator
+    ///      State Machine: before depositStake() OR before depositPayment()
+    function cancel() public {
+        // restrict access control
+        require(isSeller(msg.sender) || isBuyer(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or buyer or active operator");
+        // restrict state machine
+        require(isOpen() || onlyStakeDeposited() || onlyPaymentDeposited(), "only before deposits are completed");
+
+        // cancel escrow and return deposits
+        _cancel();
+    }
+
+    /// @notice Cancel escrow if seller does not finalize
+    ///          - return stake to seller
+    ///          - return payment to buyer
+    ///          - close escrow
+    /// @dev Access Control: buyer OR operator
+    ///      State Machine: after Countdown.isOver()
+    function timeout() public {
+        // restrict access control
+        require(isBuyer(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or active operator");
+        // restrict state machine
+        require(Countdown.isOver(), "only after countdown ended");
+
+        // cancel escrow and return deposits
+        _cancel();
```

Comment:
Should this be able to be called after `finalize()`?

Reply:
fixed in de34e9c